### PR TITLE
test: test application set up

### PR DESCRIPTION
**Pull request by xiaozhenliu-gg5**
1. Restructure test application: test_application_ts & test_application_js
2. Update circleCI set up

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -19,12 +19,24 @@ jobs:
...
-          name: Run showcase test
+          name: Run showcase test for js users
           command: |
             cp -r typescript/test/protos ./.baseline-test-out
             cd .baseline-test-out 
             npm install
-            cd ../typescript/test/test_application
+            npm pack
+            cp -r ../typescript/test/test_application_js ~/
+            cp -r showcase-0.1.0.tgz ~/test_application_js
```
Comment (alexander-fenster) [inline .circleci/config.yml]: no need to use `-r` to copy just one file

File: .circleci/config.yml [hunk 2]
```diff
@@ -19,12 +19,24 @@ jobs:
...
+            cd ~/test_application_js
+            npm install
+            npm test
+      - run:
+          name: Run showcase test for ts users
+          command: |
+            cd /home/circleci/project/.baseline-test-out
+            cp -r ../typescript/test/test_application_ts ~/
+            cp -r showcase-0.1.0.tgz ~/test_application_ts
```
Comment (alexander-fenster) [inline .circleci/config.yml]: same here

File: .circleci/config.yml (hunk 1 above)
Comment (xiaozhenliu-gg5) [reply .circleci/config.yml]: `test_application_js` is a folder, getting rid of `-r` may give error here

**Comment by xiaozhenliu-gg5** [comment]
circleCI tests fail because `#moreexports` branch in gax is deleted, blocked until new release of `google-gax`

**Comment by alexander-fenster** [comment]
Sorry! I have restored the branch. The release will happen tomorrow in the morning (automatically, when the bot wakes up).

**Comment by alexander-fenster** [comment]
I got this after restoring the branch:

image

I wonder if that means that we need to make the test application depend on `protobufjs` (it should only be a `devDependency` though).

**Comment by alexander-fenster** [comment]
Upd. I'm reading the error message incorrectly, but I don't really understand what's going on, let's take a look.

**Comment by xiaozhenliu-gg5** [comment]
checking it right now, looks a little bit wired to me, tried to delete the package-lock.json and ran it again. It actually succeeds from local.

File: typescript/test/test_application_js/package.json [hunk 3]
```diff
@@ -6,7 +6,7 @@
     "browser-test": "karma start"
   },
   "dependencies": {
-    "showcase": "file:///../../../.baseline-test-out",
+    "showcase": "file:./showcase-0.1.0.tgz",
     "@grpc/grpc-js": "^0.5.2",
     "grpc": "^1.23.3",
```
Comment (alexander-fenster) [inline typescript/test/test_application_js/package.json]: `grpc` is not used in this test application.

File: typescript/test/test_application_js/.gitignore [hunk 4]
```diff
@@ -1,3 +1,4 @@
 package-lock.json
 node_modules
 showcase-server
+index_ts.js
```
Comment (alexander-fenster) [inline typescript/test/test_application_js/.gitignore]: What is this file?

File: .circleci/config.yml (hunk 1 above)
Comment (alexander-fenster) [reply .circleci/config.yml]: I'm writing about line 29, where you copy just one file. `-r` is not needed there.

File: typescript/test/test_application_js/.gitignore (hunk 4 above)
Comment (xiaozhenliu-gg5) [reply typescript/test/test_application_js/.gitignore]: it was used to ignore the compiled file for `index_ts.ts`, removed it, thanks

File: typescript/test/test_application_js/package.json (hunk 3 above)
Comment (xiaozhenliu-gg5) [reply typescript/test/test_application_js/package.json]: removed it

File: typescript/test/test_application_ts/showcase-setup.sh [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+cd ..
+showcase-server/gapic-showcase run &
+showcase_pid=$!
+stop_showcase() {
+	kill $showcase_pid
+	# Wait for the process to die, but don't report error from the kill.
+	wait $showcase_pid || true
+}
+tsc --version
+tsc ./src/index.ts
```
[the remark this answers is not in the document]
Comment (xiaozhenliu-gg5) [reply typescript/test/test_application_ts/showcase-setup.sh]: ran `npx gts init` and disabled `npx gts check` for using `//@ts-ignore`

File: .circleci/config.yml (hunk 1 above)
Comment (xiaozhenliu-gg5) [reply .circleci/config.yml]: you are right, removed `-r`

File: .circleci/config.yml (hunk 2 above)
Comment (xiaozhenliu-gg5) [reply .circleci/config.yml]: Done, thanks

**Pull request merged**